bestiary/core/schema.py

```diff
@@ -246,9 +267,77 @@ class Meta:
     )
 
 
+class EcosystemPaginatedType(AbstractPaginatedType):
+    entities = graphene.List(EcosystemType)
+    page_info = graphene.Field(PaginationType)
+
+
+class AddEcosystem(graphene.Mutation):
+    class Arguments:
+        name = graphene.String()
+        title = graphene.String(required=False)
+        description = graphene.String(required=False)
+
+    ecosystem = graphene.Field(lambda: EcosystemType)
+
+    @check_auth
+    def mutate(self, info, name, title=None, description=None):
+        user = info.context.user
+        ctx = BestiaryContext(user)
+
+        ecosystem = add_ecosystem(ctx, name, title=title, description=description)
+
+        return AddEcosystem(
+            ecosystem=ecosystem
+        )
+
+
+class UpdateEcosystem(graphene.Mutation):
+    class Arguments:
+        ecosystem_id = graphene.Int()
+        data = EcosystemInputType()
+
+    ecosystem = graphene.Field(lambda: EcosystemType)
+
+    @check_auth
+    def mutate(self, info, ecosystem_id, data):
+        user = info.context.user
+        ctx = BestiaryContext(user)
+
+        ecosystem = update_ecosystem(ctx, ecosystem_id, **data)
+
+        return UpdateEcosystem(
+            ecosystem=ecosystem
+        )
+
+
+class DeleteEcosystem(graphene.Mutation):
+    class Arguments:
+        ecosystem_id = graphene.Int()
+
+    ecosystemName = graphene.String()
+
+    @check_auth
+    def mutate(self, info, ecosystem_id):
+        user = info.context.user
+        ctx = BestiaryContext(user)
+
+        ecosystem = delete_ecosystem(ctx, ecosystem_id)
+
+        return DeleteEcosystem(
+            ecosystemName=ecosystem.name
```

Comment:
What's the point of returning the name only? Wouldn't be better to return the same result as in `UpdateEcosystem` and `AddEcosystem`?

Reply:
I was having problems returning the complete ecosystem object as in `AddEcosystem` and `UpdateEcosystem`. It fails when you ask for the id in the returning query, as that reference is removed when the object is deleted from the database.

I agree this should be as you say, do you have any idea? Thank you in advance!

Reply:
You are not removing anything in `update` and `add`. Why are you having problems there?

Reply:
Ohh. I misunderstood what you mean. What do you mean with `when you ask for the id in the returning query`? Do you mean something like:

```
mutation delete {
    deleteEcosystem(ecosystemId: 1) {
        id
        name
    }
}
```

Reply:
Yes, that's right! My bad, I should have written an example :sweat_smile: .

Reply:
I'm sure there's a way to do it. Probably you need to set the id to none before returning it.

Reply:
By the way, if the problem is to return the ID, you can always create a `EcosystemType` object and copy the parameters there. I thinkg that might work.

Reply:
The object is returned, and the value in the `id` field is `None`. The error raises because an `EcosystemType` is returned where the `id` is null while according to the model, this is a not-nullable field.

Reply:
This has been solved by setting the former ID value to the object before returning it within the API method.